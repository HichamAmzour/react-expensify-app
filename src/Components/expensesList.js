import React from 'react';
import {connect} from 'react-redux';
import ExpenseItem from './expenseItem';
import getVisibleExpenses from '../ReduxFolder/selectors/expensesSelector';

//casue we will this this component not connected component to the store
export const ExpenseList = (props)=>(
    <div>
    <h1>Expense List</h1>
    {
        props.expenses.length === 0 ? (<p>There is no expenses in the list</p>) :
        props.expenses.map((ex)=>(
            <ExpenseItem key={ex.id} {...ex}/>
        ))
    }
    </div>
);
const mapStateToProps = (state)=>{
    return {
        expenses : getVisibleExpenses(state.expenses,state.filters)
    };
}
const ConnectToStore = connect(mapStateToProps)(ExpenseList);
export default ConnectToStore;