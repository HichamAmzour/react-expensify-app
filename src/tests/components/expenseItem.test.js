import React from 'react';
import {shallow} from 'enzyme';
import ExpenesItem from '../../Components/expenseItem';
import expensesList from '../fixturesDb/expensesList';


test("should render expense item !", ()=>{
    const wrapper= shallow(<ExpenesItem {...expensesList[0]} />)
    expect(wrapper).toMatchSnapshot()
})