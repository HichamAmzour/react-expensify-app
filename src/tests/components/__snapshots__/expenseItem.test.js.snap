// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense item ! 1`] = `
<div
  key="1"
>
  <hr />
  <h4>
    Description : 
    rent
  </h4>
  <h4>
    Amount      : 
    150
  </h4>
  <h4>
    Notes       : 
  </h4>
  <h4>
    Created At  : 
    -345600000
  </h4>
  <Link
    to="/edit/1"
  >
    Edit
  </Link>
  <hr />
</div>
`;
